service/app.py

```diff
@@ -136,6 +136,53 @@ def get_professor_data():
     # Include the full instructor name in the response
     return jsonify({"professor": full_instructor_name, "courses": course_data})
 
+@app.route('/class', methods=['GET'])
+def get_class_data():
+    class_id = request.args.get('classId') 
+    if not class_id:
+        return jsonify({"error": "Class ID is required"}), 400
+
+    # Fetch class data from the database
+    class_data = ClassData.query.filter_by(class_name=class_id).first() 
+
+    if class_data:
+        return jsonify({"class": {
+            "title": f"{class_data.subject} {class_data.class_name}",
+            "code": class_data.class_name,
+            "instructor": class_data.instructor_name,
+            "grades": {
+                "A": class_data.a,
+                "B": class_data.b,
+                "C": class_data.c,
+                "D": class_data.d,
+                "F": class_data.f,
+                "P": class_data.p,
+                "W": class_data.w,
+            },
+        }}), 200
+    else:
+        return jsonify({"error": "Class not found"}), 404
+    
+@app.route('/class/details', methods=['GET'])
+def get_class_details():
```

Comment:
I know this is prob for a separate description of the class but I think you could just add the "description" line to the above function and remove get_class_details() to eliminate some redundancy

Reply:
Even though it is redundant as it is right now, I'm going to leave it and develop it further for when we webscrape the class information.